tests/onnxruntime/test_optimization.py

```diff
@@ -322,6 +324,122 @@ def test_optimization_levels_gpu(self, test_name: str, model_arch: str, use_cach
+            device = "cuda"
+        else:
+            for_gpu = False
+            device = "cpu"
+
+        ort_model = ORTModelForSpeechSeq2Seq.from_pretrained(
+            self.onnx_model_dirs[export_name], use_cache=use_cache, provider=provider, use_io_binding=use_io_binding
+        )
+
+        optimizer = ORTOptimizer.from_pretrained(ort_model)
+
+        optimization_config = AutoOptimizationConfig.with_optimization_level(optimization_level, for_gpu=for_gpu)
+        optimization_config.disable_shape_inference = True
+        model_id = MODEL_NAMES[model_arch]
+
+        with tempfile.TemporaryDirectory(suffix="_optimized") as tmp_dir:
+            optimizer.optimize(save_dir=tmp_dir, optimization_config=optimization_config)
+
+            optimized_model = ORTModelForSpeechSeq2Seq.from_pretrained(
+                tmp_dir, use_cache=use_cache, provider=provider, use_io_binding=use_io_binding
+            )
+
+            expected_ort_config = ORTConfig(optimization=optimization_config)
+            ort_config = ORTConfig.from_pretrained(tmp_dir)
+
+            # Verify the ORTConfig was correctly created and saved
+            self.assertEqual(ort_config.to_dict(), expected_ort_config.to_dict())
+
+            data = self._generate_random_audio_data()
+            processor = get_preprocessor(model_id)
+            features = processor.feature_extractor(data, return_tensors="pt").to(device)
+
+            model_outputs = ort_model.generate(features["input_features"])
+
+            optimized_model_outputs = optimized_model.generate(features["input_features"])
+
+            self.assertTrue(torch.equal(model_outputs, optimized_model_outputs))
+            gc.collect()
+
+    @parameterized.expand(
+        grid_parameters(
+            {
+                "model_arch": SUPPORTED_ARCHITECTURES,
+                "use_cache": [False, True],
+                "optimization_level": ["O1", "O2", "O3"],
+            }
+        )
+    )
+    def test_optimization_levels_cpu(self, test_name: str, model_arch: str, use_cache: bool, optimization_level: str):
+        self._test_optimization_levels(
+            test_name=test_name,
+            model_arch=model_arch,
+            use_cache=use_cache,
+            optimization_level=optimization_level,
+            provider="CPUExecutionProvider",
+        )
+
+    @parameterized.expand(
+        grid_parameters(
+            {
+                "model_arch": SUPPORTED_ARCHITECTURES,
+                "use_cache": [True],
+                "optimization_level": ["O1", "O2", "O3", "O4"],
+            }
+        )
+    )
+    @require_torch_gpu
+    @pytest.mark.gpu_test
+    def test_optimization_levels_gpu(self, test_name: str, model_arch: str, use_cache: bool, optimization_level: str):
+        for use_io_binding in [False, True]:
+            # TODO: investigate why marian with IO Binding fails
+            if model_arch == "marian" and use_io_binding is True:
+                continue
```

Comment:
I think we can remove this as Marian is not a speech model. And move `use_io_binding` into the `grid_parameters` above.